src/graphnet/models/task/task.py

```diff
@@ -118,7 +120,18 @@ def compute_loss(self, pred: Union[Tensor, Data], data: Data) -> Tensor:
             [data[label] for label in self._target_labels], dim=1
         )
         target = self._transform_target(target)
-        loss = self._loss_function(pred, target) + self._regularisation_loss
+        if self._loss_weight_column is not None:
+            weights = data[self._loss_weight_column]
+        else:  # if no weights are given, applies weights of 1.
+            weights = (
+                torch.ones(len(pred), dtype=torch.int)
+                .reshape(-1, 1)
+                .to(pred.device)
+            )
```

Comment:
Default behaviour should be handled by `LossFunction`, I figure?
```suggestion
        else:
            weights = None
```